modules/dbparser/groupingby.c

```diff
@@ -259,6 +268,8 @@ grouping_by_expire_entry(TimerWheel *wheel, guint64 now, gpointer user_data)
             evt_tag_str("context-id", context->key.session_id),
             log_pipe_location_tag(&self->super.super.super));
 
+  if (self->sort_key)
+    correllation_context_sort(context, self->sort_key);
```

Comment:
It might worth mentioning in the documentation that sorting is done when the context is about to be closed, but before evaluating `having()` and creating the synthetic message.

(I don't know if it's possible to use message indexes in filters for example, but if so, `trigger()` will see a different order than `having()`.)

Reply:
@fekete-robert